engine/language-server/src/main/scala/org/enso/languageserver/search/SuggestionsHandler.scala

```diff
@@ -386,28 +299,40 @@ final class SuggestionsHandler(
     * @return the API suggestions database update notification
     */
   private def applyDatabaseUpdates(
-    msg: Api.SuggestionsDatabaseUpdateNotification
+    msg: Api.SuggestionsDatabaseModuleUpdateNotification
   ): Future[SuggestionsDatabaseUpdateNotification] = {
-    val (added, removed) = msg.updates
-      .foldLeft((Seq[Suggestion](), Seq[Suggestion]())) {
-        case ((add, remove), msg: Api.SuggestionsDatabaseUpdate.Add) =>
-          (add :+ msg.suggestion, remove)
-        case ((add, remove), msg: Api.SuggestionsDatabaseUpdate.Remove) =>
-          (add, remove :+ msg.suggestion)
+    val (addCmds, removeCmds, cleanCmds) = msg.updates
+      .foldLeft((Seq[Suggestion](), Seq[Suggestion](), Seq[String]())) {
+        case ((add, remove, clean), m: Api.SuggestionsDatabaseUpdate.Add) =>
+          (add :+ m.suggestion, remove, clean)
```

Comment:
This is going to be slow for larger modules, use `::` and reverse at the end.

Reply:
Same for all other branches.

Reply:
Then, it's better to use vectors here. For some reason, I decided that the default implementation of `Seq` is a `Vector`.